guardian/editions
# make use issue summary sequential

## Summary

This makes the whole fetch and update process of issue summaries sequential and safe: ie. no  parallel fetches, no overriding of files, and we use a "reducer" model where each state feeds into the next. 

This is an attempt to address this bug in a systematic way rather than trying to plug the holes: https://trello.com/c/g64so6QL/1099-sometimes-all-photos-are-blank-editions-list-shows-no-front-nothing-is-clickable

I've left a few FIXME/TODO in the comments for future possible improvements that I did not want to bundle in this changeset that is already quite large.

## Test Plan

1. Open the app, verify you get the list of editions.
2. Switch to an older edition.
3. Background then foreground the app, notice it doesn't force it back to the latest edition.
4. Change number of editions from 7 to 14, notice it refetches list of editions as expected.
